openstreetmap/josm-plugins
# 90 make circles with building tool

**Environment Setup:**
1. Follow the instructions for "Debugging in Eclipse"
https://josm.openstreetmap.de/wiki/DevelopersGuide/DevelopingPlugins
2. Make sure to remove your "JOSM" project from the building path if it's in your workspace
3. Download Building Tool Plugin

**Test**
1. Create a new data layer and select the "Draw Building" button (or push 'B')
2. Go to Data > Set Buildings Size and change shape to Circle (or once in Draw Building mode press SHIFT+C to switch to drawing circular buildings or SHIFT+R to draw rectangular building)
3. Draw a line and ensure a circle gets created with "building:yes" tag.
4. In the Building Size Dialog test:
     a. Width/diameter sets fixed diameter of circle
     b. Length steps sets how much the diameter increments by 
     c. Address dialog appropriately tags buildings
     d. Click "Advanced..." and add "building:silo" tag. Make sure circle buildings created after have tag.

Also ensure rectangle features are still intact
